module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Logic/RecordsHelperTest.php

```diff
@@ -0,0 +1,339 @@
+
+        $records = [$driver];
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->collectRequestStats($records);
+
+        $this->assertNull($result);
+    }
+
+    /**
+     * Test collectRequestStats when enableAjax is not set (defaults to true)
+     *
+     * @return void
+     */
+    public function testCollectRequestStatsWithDefaultAjaxSetting(): void
+    {
+        $config = new Config(['Authentication' => []]);
+        $loader = $this->createMock(Loader::class);
+
+        $ilsDetails = ['id' => 'record1', 'status' => 'available'];
+        $driver = $this->createMock(AbstractBase::class);
+
+        $driver->expects($this->once())
+            ->method('getExtraDetail')
+            ->with('ils_details')
+            ->willReturn($ilsDetails);
+
+        $records = [$driver];
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->collectRequestStats($records);
+
+        $this->assertIsArray($result);
+    }
+
+    /**
+     * Test collectRequestStats with empty records array
+     *
+     * @return void
+     */
+    public function testCollectRequestStatsWithEmptyRecords(): void
+    {
+        $config = new Config(['Authentication' => ['enableAjax' => true]]);
+        $loader = $this->createMock(Loader::class);
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->collectRequestStats([]);
+
+        $this->assertIsArray($result);
+    }
+
+    /**
+     * Test getDrivers maintains correct order from loadBatch
+     *
+     * @return void
+     */
+    public function testGetDriversMaintainsOrder(): void
+    {
+        $config = $this->createMock(Config::class);
+        $loader = $this->createMock(Loader::class);
+
+        $records = [
+            ['id' => 'record3', 'source' => 'Solr'],
+            ['id' => 'record1', 'source' => 'Solr'],
+            ['id' => 'record2', 'source' => 'Solr'],
+        ];
+
+        $driver1 = $this->createMock(AbstractBase::class);
+        $driver2 = $this->createMock(AbstractBase::class);
+        $driver3 = $this->createMock(AbstractBase::class);
+
+        $mockDrivers = [$driver1, $driver2, $driver3];
+
+        $loader->expects($this->once())
+            ->method('loadBatch')
+            ->willReturn($mockDrivers);
+
+        $driver1->expects($this->once())
+            ->method('setExtraDetail')
+            ->with('ils_details', $records[0]);
+
+        $driver2->expects($this->once())
+            ->method('setExtraDetail')
+            ->with('ils_details', $records[1]);
+
+        $driver3->expects($this->once())
+            ->method('setExtraDetail')
+            ->with('ils_details', $records[2]);
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->getDrivers($records);
+
+        $this->assertSame($driver1, $result[0]);
+        $this->assertSame($driver2, $result[1]);
+        $this->assertSame($driver3, $result[2]);
```

Comment:
I think this might be both more specific and more concise, if it works:

```suggestion
        $this->assertSame([$driver1, $driver2, $driver3], $result);
```